app/services/config_data_sources/json_project_data_source.rb

```diff
@@ -147,7 +151,15 @@ def job_triggers_from_hash_array(job_trigger_array: nil)
 
     def projects=(projects)
       JSONProjectDataSource.projects_file_semaphore.synchronize do
-        File.write(git_repo.file_path("projects.json"), JSON.pretty_generate(projects.map(&:to_object_dictionary)))
+        # We have to ignore the `environment_variables` instance variable, as it's stored in a separate file
+        json_data = JSON.pretty_generate(projects.map do |project|
```

Comment:
I would prefer to have this be a 2 step process, first is `projects.map`, then second be the `JSON.pretty_generate`

Reply:
Yeah, that's valid, will update tomorrow